tests/_dockerfiles/ubuntu-16.04/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:16.04
+ARG AWK
+ARG NODE_VERSION=16
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get -y update && \
+    apt-get -y install \
+        --no-install-recommends \
+        ca-certificates \
+        curl \
+        gawk \
+        git \
+        jq \
+        php \
+        zsh && \
+    update-alternatives --set awk "/usr/bin/$AWK" && \
+    curl -fsSL "https://deb.nodesource.com/setup_$NODE_VERSION.x" | bash - && \
+    curl -fsSL 'https://dl.yarnpkg.com/debian/pubkey.gpg' | gpg --dearmor > /usr/share/keyrings/yarnkey.gpg && \
+    echo 'deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main' > /etc/apt/sources.list.d/yarn.list && \
```

Comment:
I am not familiar with the package management of Ubuntu, but can't they be placed before the first `apt-get -y update` (line 5)?

Reply:
The ubuntu:16.04 has neither curl nor ca-certificates installed, hence the repository set ups need to run after curl and ca-certificates become available; otherwise, they would simply fail.